src/state/selectors/legendSelectors.ts

```diff
@@ -10,5 +11,5 @@ const selectAllLegendPayload2DArray = (state: RechartsRootState): ReadonlyArray<
 
 export const selectLegendPayload: (state: RechartsRootState) => ReadonlyArray<LegendPayload> = createSelector(
   [selectAllLegendPayload2DArray],
-  payloads => payloads.flat(1),
+  payloads => sortBy(payloads.flat(1), 'value'),
```

Comment:
Maybe we should start using native sort functions? But.. maybe not. Just throwing it out there. Or maybe es-toolkit

People get upset about lodash for whatever silly reason.

Reply:
Meh. As long as lodash is in recharts people will complain about bundle size. I have no preference so all good